lib/inspec/resources/sys_info.rb

```diff
@@ -3,117 +3,140 @@
-      else
-        skip_resource "The `sys_info.manufacturer` resource is not supported on your OS yet."
-      end
+      @cache.info[:manufacturer]
     end
 
-    # returns the ServerModel of the local system
     def model
-      os = inspec.os
-      if os.darwin?
-        inspec.command("sysctl -n hw.model").stdout.chomp
-      elsif os.linux?
-        inspec.command("cat /sys/class/dmi/id/product_name").stdout.chomp
-      elsif os.windows?
-        inspec.powershell("Get-CimInstance -ClassName Win32_ComputerSystem | Select Model -ExpandProperty Model").stdout.chomp
-      else
-        skip_resource "The `sys_info.model` resource is not supported on your OS yet."
-      end
+      @cache.info[:model]
+    end
+
+    def totalmemory_kb
+      @cache.info[:totalmemory_kb]
+    end
+
+    def hostname
+      @cache.info[:hostname]
+    end
+
+    def fqdn
+      @cache.info[:fqdn]
     end
 
     def to_s
       "System Information"
     end
   end
+
+  class LinuxInfo < SysInfo
+    attr_reader :inspec
+    def initialize(inspec)
+      @inspec = inspec
+    end
+
+    def info
+      fqdn = inspec.command("hostname -f").stdout.chomp
+      domain = inspec.command("hostname -d").stdout.chomp
+      ip = inspec.command("hostname -I").stdout.chomp
+      short = inspec.command("hostname -s").stdout.chomp
+      manufacturer = inspec.command("cat /sys/class/dmi/id/sys_vendor").stdout.chomp
+      model = inspec.command("cat /sys/class/dmi/id/product_name").stdout.chomp
+      totalmemory_kb = inspec.command("awk '/MemTotal/ {print $2}' /proc/meminfo").stdout.chomp
+
+      {
+        fqdn: fqdn,
+        domain: domain,
+        ip: ip,
+        hostname: short,
+        manufacturer: manufacturer,
+        model: model,
+        totalmemory_kb: totalmemory_kb,
+      }
+    end
+  end
+
+  class WindowsInfo < SysInfo
+    attr_reader :inspec
+    def initialize(inspec)
+      @inspec = inspec
+    end
+
+    def info
+      cmd = inspec.command <<-EOF
+        Get-WmiObject Win32_ComputerSystem -Property * | convertto-json
+      EOF
+
+      cmd = {} if cmd.stdout == ""
+      begin
+        info = JSON.parse(cmd.stdout)
+      rescue JSON::ParserError => _e
+        return nil
+      end
+
+      ip_cmd = inspec.command("(Get-WmiObject Win32_NetWorkAdapterConfiguration | Where IPAddress -ne $null | select -Property ipaddress).IPAddress | ConvertTo-Json")
+      ip_cmd = {} if ip_cmd.stdout == ""
+      begin
+        ip_info = JSON.parse(ip_cmd.stdout)
+      rescue JSON::ParserError => _e
+        return nil
+      end
+
+      if (info["DomainRole"] == 2 || info["DomainRole"] == 0)
+        fqdn = info["DNSHostName"]
+        domain = nil
+      else
+        fqdn = "#{info["DNSHostName"]}.#{info["Domain"]}"
+        domain = info["Domain"]
+      end
+
+      memory = info["TotalPhysicalMemory"] / 1024 rescue nil
```

Comment:
I'd prefer `info.fetch("TotalPhysicalMemory", 0) / 1024` to resorting to exception handling here.

Reply:
sorry for my stupid Questions. 
Will info.fetch("TotalPhysicalMemory",0) / 1024 not raise an divion by Zero error?
In case TotalPhysicalMemory can not be found as key it will Reply the value as 0 and then try to divide it through 1024?
Or do i missunderstand this function?

Reply:
0/1 is always 0. 1/0 is undefined/error.

Reply:
yah my fault. already figured that out ;)